proxy/forward_auth.go

```diff
@@ -47,7 +68,9 @@ func (p *Proxy) nginxCallback(w http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
-func (p *Proxy) traefikCallback(w http.ResponseWriter, r *http.Request) error {
+// forwardedURIHeaderCallback handles the post-authentication callback from
+// forwarding proxies that support the `X-Forwarded-Uri` header per rfc7239.
```

Comment:
rfc7239 seems to specify a standard `Forwarded` header which has all the `X-Forwarded-*` headers merged together:

```
Forwarded: for=192.0.2.60;proto=http;by=203.0.113.43
```

I wasn't able to find a spec detailing the `X-` style headers. Maybe we should note they're non-standard?

Reply:
You are exactly right, I had this backwards. Looks like x-* style headers are actually deprecated but still wildly used. 

https://tools.ietf.org/html/rfc6648

https://stackoverflow.com/a/3561399